### PR TITLE
Add document:mReplace

### PR DESCRIPTION
**Pull request by Yoann-Abbes**
## What does this PR do ?

This PR implements the `mReplace` method with its unit tests.

### How should this be manually tested?

Clone this branch and run unit tests
`./gradlew test`

When it succeed, compile it

`./gradlew jar`

Initiate another java project by adding the compiled SDK as a dependency.

Then, run Kuzzle, create an index `nyc-open-data` and a `yellow-taxi` collection in the admin console. Create 2 documents `some-id1` and `some-id2`.
Finally, run this code

```java
import io.kuzzle.sdk.*;
import io.kuzzle.sdk.Options.KuzzleOptions;
import io.kuzzle.sdk.Options.Protocol.WebSocketOptions;
import io.kuzzle.sdk.Protocol.WebSocket;

import java.util.concurrent.ConcurrentHashMap;

public class mCreateOrReplaceDocument {
    private static Kuzzle kuzzle;

    public static void main(String[] args) {
        WebSocketOptions opts = new WebSocketOptions();
        opts.setAutoReconnect(true).setConnectionTimeout(42000);

        try {
            WebSocket ws = new WebSocket("localhost", opts);

            kuzzle = new Kuzzle(ws, (KuzzleOptions) null);

            kuzzle.connect();

            ConcurrentHashMap<String, Object> document1 = new ConcurrentHashMap<>();
            ConcurrentHashMap<String, Object> body1 = new ConcurrentHashMap<>();

            ConcurrentHashMap<String, Object> document2 = new ConcurrentHashMap<>();
            ConcurrentHashMap<String, Object> body2 = new ConcurrentHashMap<>();

            document1.put("_id", "some-id1");
            body1.put("key", "value");
            document1.put("body", body1);

            document2.put("_id", "some-id2");
            body2.put("key", "value");
            document2.put("body", body2);

            ArrayList<ConcurrentHashMap<String, Object>> documents = new ArrayList<>();
            documents.add(document1);
            documents.add(document2);

            ConcurrentHashMap<String, Object> response =
            kuzzle.getDocumentController().mReplace("nyc-open-data", "yellow-taxi", documents)
            .get();

            System.out.println(response);
        }  catch (Exception e) {
            e.printStackTrace();
        }

        kuzzle.disconnect();
    }
};
```

You should see the documents replaced in your collection.

### Other changes

Update subscribe documentation -> Remove non existing snippets

### CODE REVIEW

File: doc/3/controllers/realtime/subscribe/index.md [hunk 1]
```diff
@@ -51,11 +51,3 @@ _Simple subscription to document notifications_
 _Subscription to document notifications with scope option_
 
 <<< ./snippets/document-notifications-leave-scope.java
-
-_Subscription to message notifications_
```
Comment (scottinet) [inline doc/3/controllers/realtime/subscribe/index.md]: why is this removed?

Comment (Yoann-Abbes) [reply doc/3/controllers/realtime/subscribe/index.md]: Those lines are a mistake from the realtime controller PR, those snippets do not exists
